### PR TITLE
Psp 1786 access request dev fixes.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: backend/dal/Configuration/AccessRequestConfiguration.cs [hunk 1]
```diff
@@ -35,8 +34,8 @@ public override void Configure(EntityTypeBuilder<AccessRequest> builder)
             builder.HasOne(m => m.Status).WithMany(u => u.AccessRequests).HasForeignKey(m => m.StatusId).OnDelete(DeleteBehavior.Cascade).HasConstraintName("PIM_ARQSTT_PIM_ACRQST_FK");
 
             builder.HasMany(m => m.Organizations).WithMany(m => m.AccessRequests).UsingEntity<AccessRequestOrganization>(
-                m => m.HasOne(m => m.Organization).WithMany(m => m.AccessRequestsManyToMany).HasForeignKey(m => m.OrganizationId),
-                m => m.HasOne(m => m.AccessRequest).WithMany(m => m.OrganizationsManyToMany).HasForeignKey(m => m.AccessRequestId)
+                m => m.HasOne(m => m.Organization).WithMany(m => m.AccessRequestsManyToMany).HasForeignKey(m => m.OrganizationId).OnDelete(DeleteBehavior.ClientCascade),
```
Comment (Fosol) [inline backend/dal/Configuration/AccessRequestConfiguration.cs:37]: Are these foreign keys named correctly?

File: backend/dal/Services/AccessRequest.cs [hunk 2]
```diff
@@ -189,10 +197,12 @@ public AccessRequest Update(AccessRequest update)
             // Copy values into entity.
             accessRequest.RoleId = update.RoleId;
             accessRequest.RowVersion = update.RowVersion;
```
Comment (Fosol) [inline backend/dal/Services/AccessRequest.cs:199]: Make sure concurrency works.  I know just setting the value like this normally does not.

File: backend/entities/AccessRequestStatusType.cs [hunk 3]
```diff
@@ -38,4 +38,15 @@ public AccessRequestStatusType(string id, string description) : base(id, descrip
         }
         #endregion
     }
+
+    public static class AccessRequestStatusTypes
```
Comment (Fosol) [inline backend/entities/AccessRequestStatusType.cs:42]: Generally bad practice to hide a **public** class in another's file.

**Review by Fosol — APPROVED** [review]

**Comment by Fosol** [comment]
Seemed to be a lot of places the URL needed to be changed.  Makes me wonder if we are not centralizing the api correctly.

File: backend/entities/AccessRequestStatusType.cs (hunk 3 above)
Comment (devinleighsmith) [reply backend/entities/AccessRequestStatusType.cs:42]: It is good practice to co-locate related functionality, such as a static constants class that is intended for use with the sibling class.

**Comment by devinleighsmith** [comment]
The url needed to be changed a lot of places for two reasons:
1) some of the controllers were incorrectly appending duplicate information on top of a base route. IE: /access/requests/access/requests
2) there are tests written to explicitly check that individual controller methods have specific routes. My initial attempt to place all of a controllers routing information at the top level route caused these tests to fail.

File: backend/dal/Configuration/AccessRequestConfiguration.cs (hunk 1 above)
Comment (devinleighsmith) [reply backend/dal/Configuration/AccessRequestConfiguration.cs:37]: PIM_ACRQST_PIM_ACRQOR_FK and PIM_ORG_PIM_ACRQOR_FK

**Pull request merged**